app/ide-desktop/lib/content/src/index.ts

```diff
@@ -753,117 +753,51 @@ function ok(value: any) {
-        this.dark_theme = ok(other.dark_theme) ? tryAsBoolean(other.dark_theme) : this.dark_theme
-        this.high_contrast = ok(other.high_contrast)
-            ? tryAsBoolean(other.high_contrast)
-            : this.high_contrast
-        this.use_loader = ok(other.use_loader) ? tryAsBoolean(other.use_loader) : this.use_loader
-        this.wasm_url = ok(other.wasm_url) ? tryAsString(other.wasm_url) : this.wasm_url
-        this.wasm_glue_url = ok(other.wasm_glue_url)
-            ? tryAsString(other.wasm_glue_url)
-            : this.wasm_glue_url
-        this.node_labels = ok(other.node_labels)
-            ? tryAsBoolean(other.node_labels)
-            : this.node_labels
-        this.crash_report_host = ok(other.crash_report_host)
-            ? tryAsString(other.crash_report_host)
-            : this.crash_report_host
-        this.data_gathering = parseBoolean(other.data_gathering) ?? this.data_gathering
-        this.is_in_cloud = ok(other.is_in_cloud)
-            ? tryAsBoolean(other.is_in_cloud)
-            : this.is_in_cloud
-        this.verbose = ok(other.verbose) ? tryAsBoolean(other.verbose) : this.verbose
-        this.authentication_enabled = ok(other.authentication_enabled)
-            ? tryAsBoolean(other.authentication_enabled)
-            : this.authentication_enabled
-        this.email = ok(other.email) ? tryAsString(other.email) : this.email
-        this.application_config_url = ok(other.application_config_url)
-            ? tryAsString(other.application_config_url)
-            : this.application_config_url
-        this.test_workflow = ok(other.test_workflow)
-            ? tryAsString(other.test_workflow)
-            : this.test_workflow
-        this.skip_min_version_check = ok(other.skip_min_version_check)
-            ? tryAsBoolean(other.skip_min_version_check)
-            : this.skip_min_version_check
-        this.preferred_engine_version =
-            semver.parse(other.preferred_engine_version) ?? this.preferred_engine_version
-        this.enable_new_component_browser =
-            parseBoolean(other.enable_new_component_browser) ?? this.enable_new_component_browser
-    }
-}
-
-function parseBoolean(value: any): boolean | null {
-    if (value === 'true' || value === true) {
-        return true
-    } else if (value === 'false' || value === false) {
-        return false
-    } else {
-        return null
+        for (let key of Object.keys(this)) {
+            let self: any = this
+            let otherVal = other[key]
+            let selfVal = self[key]
+            if (ok(otherVal)) {
+                if (typeof selfVal === 'boolean') {
+                    self[key] = tryAsBoolean(otherVal)
+                } else if (selfVal instanceof SemVer) {
+                    self[key] = semver.parse(otherVal)
+                } else {
+                    self[key] = tryAsString(otherVal)
+                }
+            }
+        }
```

Comment:
Have you tested each argument that it's still working? I see for example, that the "enable_new_component_browser" or "data_gathering" uses parseBoolean, not tryAsBoolean. It may affect how those variables are passed as URL path or CLI argument.